File: src/main/scala/codechicken/multipart/handler/CubicChunksCompat.scala
package codechicken.multipart.handler

import com.cardinalstar.cubicchunks.api.event.CubeEvent
import cpw.mods.fml.common.eventhandler.{EventPriority, SubscribeEvent}
import net.minecraftforge.common.MinecraftForge
import scala.collection.JavaConversions._

object CubicChunksCompat {
  def init(): Unit = MinecraftForge.EVENT_BUS.register(this)

  @SubscribeEvent(priority = EventPriority.HIGHEST)
  def onTileEntitiesLoad(event: CubeEvent.DataLoad): Unit = {
    val cube = event.cube
    val world = event.world

    MultipartSaveLoad.loadTiles(world, cube.cubeTileEntityMap)
  }
}

File: src/main/scala/codechicken/multipart/handler/CubicChunksCompat.java
package codechicken.multipart.handler;

import net.minecraftforge.common.MinecraftForge;

import com.cardinalstar.cubicchunks.api.event.CubeEvent;

import cpw.mods.fml.common.eventhandler.EventPriority;
import cpw.mods.fml.common.eventhandler.SubscribeEvent;

public final class CubicChunksCompat {

    private static final CubicChunksCompat INSTANCE = new CubicChunksCompat();

    private CubicChunksCompat() {}

    public static void init() {
        MinecraftForge.EVENT_BUS.register(INSTANCE);
    }

    @SubscribeEvent(priority = EventPriority.HIGHEST)
    public void onTileEntitiesLoad(CubeEvent.DataLoad event) {
        MultipartSaveLoad.loadTiles(event.world, event.cube.cubeTileEntityMap);
    }
}
